api/appsproxy/design.go

```diff
@@ -0,0 +1,275 @@
+	_ "github.com/keboola/keboola-as-code/internal/pkg/service/common/goaextension/errormsg"
+	_ "github.com/keboola/keboola-as-code/internal/pkg/service/common/goaextension/genericerror"
+	_ "github.com/keboola/keboola-as-code/internal/pkg/service/common/goaextension/oneof"
+	_ "github.com/keboola/keboola-as-code/internal/pkg/service/common/goaextension/operationid"
+	. "github.com/keboola/keboola-as-code/internal/pkg/service/common/goaextension/token"
+)
+
+// API definition ------------------------------------------------------------------------------------------------------
+
+// nolint: gochecknoinits
+func init() {
+	dependenciesType := func(method *service.MethodData) string {
+		if dependencies.HasSecurityScheme("APIKey", method) {
+			return "dependencies.ProjectRequestScope"
+		}
+		return "dependencies.PublicRequestScope"
+	}
+	dependencies.RegisterPlugin(dependencies.Config{
+		Package:            "github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/dependencies",
+		DependenciesTypeFn: dependenciesType,
+		DependenciesProviderFn: func(method *service.EndpointMethodData) string {
+			t := dependenciesType(method.MethodData)
+			return fmt.Sprintf(`ctx.Value(%sCtxKey).(%s)`, t, t)
+		},
+	})
+}
+
+var _ = API("appsproxy", func() {
+	Randomizer(expr.NewDeterministicRandomizer())
+	Title("Data application proxy")
+	Description("A service for proxing requests/authorization to data applications using Keboola components.")
+	Version("1.0")
+	HTTP(func() {
+		Path("/_proxy/api/v1")
+		Consumes("application/json")
+		Produces("application/json")
+	})
+	Server("appsproxy", func() {
+		Host("production", func() {
+			URI("https://appsproxy.{stack}")
+			Variable("stack", String, "Base URL of the stack", func() {
+				Default("keboola.com")
+				Enum("keboola.com", "eu-central-1.keboola.com", "north-europe.azure.keboola.com", "eu-west-1.aws.keboola.dev", "east-us-2.azure.keboola-testing.com")
+			})
+		})
+		Host("localhost", func() {
+			URI("http://localhost:8000")
+		})
+	})
+})
+
+// Service definition --------------------------------------------------------------------------------------------------
+
+var _ = Service("apps-proxy", func() {
+	Description("Service for proxing requests/authorization to data applications using Keboola data app component.")
+	// CORS
+	cors.Origin("*", func() {
+		cors.Headers("Content-Type", "X-StorageApi-Token")
+		cors.Methods("GET", "POST", "PUT", "DELETE")
+	})
+
+	// Set authentication by token to all endpoints without NoSecurity()
+	Security(tokenSecurity)
+	defer AddTokenHeaderToPayloads(tokenSecurity, "storageApiToken", "X-StorageApi-Token")
+
+	// Auxiliary endpoints ---------------------------------------------------------------------------------------------
+
+	Method("ApiRootIndex", func() {
+		Meta("openapi:summary", "Redirect to /_proxy")
+		Description("Redirect to /_proxy.")
+		NoSecurity()
+		HTTP(func() {
+			// Redirect / -> /_proxy
+			GET("//_proxy/api/")
```

Comment:
I cannot find it in the documentation but I would expect that it trails single slash at the start. So the resulting address would be `_proxy/api/` If I do not have double slashes at the start. You can test it.

Reply:
When I omit the first `/` it does not work, and the redirect from `/_proxy/api` returns `404`, instead of redirection. I think it is due to wrong match of the `GET` path in the design.